From bfe67e8ba14117ff774ecdc620ad4d7e0020036d Mon Sep 17 00:00:00 2001
From: Timo Rothenpieler <timo.rothenpieler@uni-bremen.de>
Date: Fri, 27 Aug 2021 15:21:46 +0200
Subject: [PATCH] Prevent automagical detection of Lua

Abort if Lua support has been requested but Lua cannot be found with
pkg-config, and use the Lua pkg-config module set up by the
eclass-provided wrapper (or unslotted Lua) instead of enumerating
installed versions.
---
 auxdir/x_ac_lua.m4 | 31 +++++++++++++++----------------
 1 file changed, 15 insertions(+), 16 deletions(-)

diff --git a/auxdir/x_ac_lua.m4 b/auxdir/x_ac_lua.m4
index b3c7c62e06..129703f308 100644
--- a/auxdir/x_ac_lua.m4
+++ b/auxdir/x_ac_lua.m4
@@ -12,20 +12,21 @@
 
 AC_DEFUN([X_AC_LUA],
 [
-	x_ac_lua_pkg_name="lua"
-	#check for 5.4, 5.3, 5.2 and then 5.1
-	PKG_CHECK_EXISTS([lua5.4], [x_ac_lua_pkg_name=lua5.4],
-		[PKG_CHECK_EXISTS([lua-5.4], [x_ac_lua_pkg_name=lua-5.4],
-		[PKG_CHECK_EXISTS([lua5.3], [x_ac_lua_pkg_name=lua5.3],
-		[PKG_CHECK_EXISTS([lua-5.3], [x_ac_lua_pkg_name=lua-5.3],
-		[PKG_CHECK_EXISTS([lua5.2], [x_ac_lua_pkg_name=lua5.2],
-		[PKG_CHECK_EXISTS([lua-5.2], [x_ac_lua_pkg_name=lua-5.2],
-		[PKG_CHECK_EXISTS([lua5.1], [x_ac_lua_pkg_name=lua5.1],
-		[PKG_CHECK_EXISTS([lua-5.1], [x_ac_lua_pkg_name=lua-5.1],
-	        [x_ac_lua_pkg_name="lua >= 5.1"])])])])])])])])
-	PKG_CHECK_MODULES([lua], ${x_ac_lua_pkg_name},
-                [x_ac_have_lua="yes"],
-                [x_ac_have_lua="no"])
+	AC_MSG_CHECKING([whether to enable support for job scripts written in Lua])
+	AC_ARG_ENABLE(
+		[lua],
+		AS_HELP_STRING(--enable-lua, enable support for job scripts written in Lua),
+		[x_ac_enable_lua="$enableval"],
+		[x_ac_enable_lua=yes])
+	AC_MSG_RESULT([$x_ac_enable_lua])
+
+	x_ac_have_lua=no
+	if test "x$x_ac_enable_lua" = "xyes"; then
+		x_ac_lua_pkg_name="lua"
+		PKG_CHECK_MODULES([lua], ${x_ac_lua_pkg_name},
+			[x_ac_have_lua="yes"],
+			[AC_MSG_ERROR([Lua support requested but Lua not found])])
+	fi
 
 	if test "x$x_ac_have_lua" = "xyes"; then
 	  saved_CFLAGS="$CFLAGS"
@@ -54,8 +55,6 @@ AC_DEFUN([X_AC_LUA],
 	  fi
 	  CFLAGS="$saved_CFLAGS"
 	  LIBS="$saved_LIBS"
-	else
-	  AC_MSG_WARN([unable to locate lua package])
 	fi
 
 	AM_CONDITIONAL(HAVE_LUA, test "x$x_ac_have_lua" = "xyes")
-- 
2.31.1

